src/main/java/com/iexec/core/chain/adapter/BlockchainAdapterClientConfig.java

```diff
@@ -20,42 +20,42 @@
 import com.iexec.blockchain.api.BlockchainAdapterApiClientBuilder;
 import com.iexec.blockchain.api.BlockchainAdapterService;
 import feign.Logger;
-import lombok.Data;
-import org.springframework.beans.factory.annotation.Value;
+import jakarta.validation.constraints.NotEmpty;
+import lombok.Value;
+import org.hibernate.validator.constraints.URL;
 import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.context.annotation.Bean;
+import org.springframework.validation.annotation.Validated;
 
 import java.time.Duration;
 
-@Data
+@Value
+@Validated
 @ConfigurationProperties(prefix = "blockchain-adapter")
 public class BlockchainAdapterClientConfig {
 
     // TODO add configuration parameters for next major version
     public static final int WATCH_PERIOD_SECONDS = 2;
     public static final int MAX_ATTEMPTS = 25;
 
-    private final String protocol;
-    private final String host;
-    private final int port;
-    // TODO improve property names before next major version
-    @Value("${blockchain-adapter.user.name}")
-    private final String username;
-    @Value("${blockchain-adapter.user.password}")
-    private final String password;
-
-    private String buildHostUrl(String protocol, String host, int port) {
-        return protocol + "://" + host + ":" + port;
-    }
+    @URL(message = "URL must be a valid URL")
+    @NotEmpty(message = "URL must not be empty")
+    String url;
+
+    BlockchainAdapterAuth auth;
 
     @Bean
     public BlockchainAdapterApiClient blockchainAdapterClient() {
         return BlockchainAdapterApiClientBuilder.getInstanceWithBasicAuth(
-                Logger.Level.NONE, buildHostUrl(protocol, host, port), username, password);
+                Logger.Level.NONE, url, auth.username, auth.password);
```

Comment:
Isn't it better to use methods instead of directly accessing fields ?
```suggestion
                Logger.Level.NONE, url, auth.username(), auth.password());
```